share/p2p/shrexnd/client.go

```diff
@@ -51,10 +50,14 @@ func NewClient(params *Parameters, host host.Host) (*Client, error) {
 func (c *Client) RequestND(
 	ctx context.Context,
 	root *share.Root,
-	nID namespace.ID,
+	namespace share.Namespace,
 	peer peer.ID,
 ) (share.NamespacedShares, error) {
-	shares, err := c.doRequest(ctx, root, nID, peer)
+	if err := namespace.ValidateDataNamespace(); err != nil {
```

Comment:
unrelated to this PR but just a thought - I wish this method were called `ValidateData` since it's called on namespace so feels redundant to have Namespace at end.

Reply:
That's true, and I was debating myself about this. I landed on adding Namespace because, ValidateData feels like we tell Namespace what to do - to validate some data.

Reply:
ValidateDataNamespace is not a good name -- even ValidateNamespaceData is better but still i think ValidateData is best

Reply:
Hmm, let's do ValidateNamespaceData

Reply:
In sync decided for ValidateForBlob and ValidateForData